src/main/webapp/app/entities/file-details.model.ts

```diff
@@ -0,0 +1,18 @@
+export class FileDetails {
+    constructor(
+        public name: string = '',
+        public extension: string = '',
+    ) {}
+    static getFileDetailsFromPath(filePath: string): FileDetails {
```

Comment:
```suggestion
    static getFileDetailsFromPath(filePath: string | undefined): FileDetails {
```
You are checking in the next line if `filePath` is not there. The typing should reflect that this can be possible.